src/poetry/installation/executor.py

```diff
@@ -855,3 +876,11 @@ def _get_archive_info(self, package: Package) -> dict[str, Any]:
             archive_info["hashes"] = {algorithm: value}
 
         return archive_info
+
+    def _serialize_operations(
+        self,
+        serial_operations: list[Operation],
+    ) -> None:
+        """Execute operations serialy"""
+        for operation in serial_operations:
+            self._execute_operation(operation)
```

Comment:
I'm sorry for being unclear in https://github.com/python-poetry/poetry/pull/9658#discussion_r1776862695. I meant: move `_serialize` out of the loop, but still inside the `execute` function.

I am not convinced that making `_serialize` a member function provides value. It's a helper function very closely coupled with the parallel execution happening in `execute`.

Reply:
Ah, I misinterpreted you. Fixed!